### PR TITLE
🌱OwnerMachine lookup fix while upgrade

### LINKED ISSUE #3143: util.GetOwnerMachine needs to exclude API version
**What steps did you take and what happened:**
1. Deploy v1alpha2
1. Create a Cluster, Machine, InfraMachine (e.g. AWSMachine)
1. Upgrade to v1alpha3
1. Try to delete the InfraMachine
1. If the controller is using util.GetOwnerMachine(), it returns nil instead of the Machine

**What did you expect to happen:**
1. Return the owning Machine

**Anything else you would like to add:**
This is because util.GetOwnerMachine is looking for an exact match on Group, Version, and Kind. But because the InfraMachine's owner reference was created with v1alpha2, the APIVersion of the owner reference is cluster.x-k8s.io/v1alpha2.

We need to adjust util.GetOwnerMachine to only look at the Group and Kind, and exclude Version.

**Environment:**

- Cluster-api version: 0.3.6

/kind bug

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:
OwnerMachine lookup fix while upgrading

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3143

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func ClusterToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToReque
 // GetOwnerMachine returns the Machine object owning the current resource.
 func GetOwnerMachine(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*clusterv1.Machine, error) {
 	for _, ref := range obj.OwnerReferences {
-		if ref.Kind == "Machine" && ref.APIVersion == clusterv1.GroupVersion.String() {
+		if ref.Kind == "Machine" && strings.Split(ref.APIVersion, "/")[0] == clusterv1.GroupVersion.Group {
```
Comment (vincepri) [inline util/util.go:310]: We might want to consider using https://github.com/kubernetes/apimachinery/blob/76330795f827e44ffc47f4fdc8e0d39872bead8f/pkg/runtime/schema/group_version.go#L211 to parse the GroupVersion and compare the groups afterwards

Comment (ykakarap) [reply util/util.go]: Thanks for the tip @vincepri 😄 

Changed to use `schema.ParseGroupVersion`

**Comment by vincepri** [comment]
LGTM, can we add a simple test?

**Comment by ykakarap** [comment]
@vincepri I added a simple test to check for a machine lookup from a different version.

Note that I was not able to use `clusterapiv1alpha2.GroupVersion.String()` because it is causing a cyclic import.

**Pull request merged**